envoy/buffer/buffer.h

```diff
@@ -109,6 +114,20 @@ class BufferMemoryAccount {
    * @param amount the amount to credit.
    */
   virtual void credit(uint64_t amount) PURE;
+
+  /**
+   * Clears the associated downstream with this account.
+   * After this has been called, calls to reset the downstream become no-ops.
+   */
+  virtual void clearDownstream() PURE;
+
+  /**
+   * Reset the downstream stream associated with this account. Resetting the downstream stream
+   * should trigger a reset of the corresponding upstream stream if it exists.
+   *
+   * @param reason the reason for reseting the stream.
+   */
+  virtual void resetDownstream(Http::StreamResetReason reason) PURE;
```

Comment:
nit: reset reason could be omitted, resetting via this interface implies that the termination is related to overload.